concordium-consensus/src/Concordium/GlobalState/Persistent/LMDB.hs

```diff
@@ -513,16 +524,35 @@ resizeDatabaseHandlers dbh size = do
     logEvent LMDB LLDebug $ "Resizing database from " ++ show oldMapSize ++ " to " ++ show newMapSize
     liftIO . withWriteStoreEnv (dbh ^. storeEnv) $ flip mdb_env_set_mapsize newMapSize
 
+-- | Load a block and its state hash (if available).
+-- Normal blocks already contain their state hash. For genesis blocks, the state hash is loaded
+-- from the metadata table if it is present there.
+loadBlock ::
```

Comment:
Shouldn't it always be present given that you are setting it when calling `loadSkovPersistentData` ? Maybe that `Maybe` is not actually required?